File: Backend/KaskoManagementSystem/Kasko.Entities/Concrete/Policy.cs
using Kasko.Entities.Abstract;
using Kasko.Entities.Enums;

namespace Kasko.Entities.Concrete
{
    public class Policy : BaseEntity
    {
        public Guid CustomerId { get; set; }
        public Guid VehicleId { get; set; }
        public Guid QuoteId { get; set; }

        public string PolicyNumber { get; set; } = string.Empty;

        public decimal PremiumAmount { get; set; }

        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }

        public PolicyStatus Status { get; set; }

        public byte[] RowVersion { get; set; } = Array.Empty<byte>();

        public virtual Customer Customer { get; set; } = null!;
        public virtual Vehicle Vehicle { get; set; } = null!;
        public virtual Quote Quote { get; set; } = null!;
    }
}

File: Fronted/KaskoManagementSystemFronted/src/app/features/policies/policy.ts
export enum PolicyStatus {
  Draft = 1,
  Active = 2,
  Expired = 3,
  Cancelled = 4
}

export interface Policy {
  id: string;
  customerId: string;
  vehicleId: string;
  quoteId: string;
  policyNumber: string;
  premiumAmount: number;
  startDate: string;
  endDate: string;
  status: PolicyStatus;
  isActive: boolean;
  rowVersion?: string;
}

export interface PolicyCreateDto {
  customerId: string;
  vehicleId: string;
  quoteId: string;
  startDate: string;
  endDate: string;
}

export interface PolicyUpdateDto {
  endDate: string;
  rowVersion: string;
}